### ISSUE TITLE
Brackets in name causing problems downloading

### ISSUE THREAD

**Issue by darchitectlp** [Vazkii/CMPDL#37]
Looks like the brackets are causing problems, too. Just tried to download PrioR and loads of issues.

https://minecraft.curseforge.com/projects/prior

Everything that uses a bracket fails. 

Do you refuse to use anything but Vaz's wonderful downloader? Here's a work around until he gets around to fixing it:

Find the errored section:
Downloading 228027/2456045
Project URL is http://minecraft.curseforge.com/projects/228027
File download URL is https://minecraft.curseforge.com/projects/bibliocraft/files/2456045/download
Downloading BiblioCraft%5bv2.2.5%5d%5bMC1.10.2%5d.jar
Error: class java.io.IOException: Server returned HTTP response code: 403 for URL: https://media.forgecdn.net/files/2456/45/BiblioCraft%255bv2.2.5%255d%255bMC1.10.2%255d.jar
sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1876)
sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1474)
sun.net.www.protocol.https.HttpsURLConnectionImpl.getInputStream(HttpsURLConnectionImpl.java:254)
java.net.URL.openStream(URL.java:1045)
vazkii.cmpdl.CMPDL.downloadFileFromURL(CMPDL.java:369)
vazkii.cmpdl.CMPDL.downloadFile(CMPDL.java:325)
vazkii.cmpdl.CMPDL.downloadModpackFromManifest(CMPDL.java:218)
vazkii.cmpdl.CMPDL.downloadFromURL(CMPDL.java:107)
vazkii.cmpdl.OperatorThread.run(OperatorThread.java:19)

Now, go the page download page of the failed mod (http://minecraft.curseforge.com/projects/228027)
Manually download.
Go to your CMPDL download folder and drag your clean, full download into it. Copy the wonky name (in this case BiblioCraft%255bv2.2.5%255d%255bMC1.10.2%255d.jar) and paste it into the legit manual download. Trash the CMPDL download. Start CMPDL again.

It will comb through everything and asses the wonky named file as a successful download and will move on to the next!

Granted, you'll have to download several files manually, but better than 150 manual downloads!

Thanks again, Vazkii, for your amazing mods.

**Comment by Rencer** [comment]
Thanks for the detailed "how-to". It works, but a bit too much work sometimes.

All those ASCII encodes are just causing problems, and I mean not just in this program, but even in everyday surfing around. Try to copy some links, and you know what I mean.

I'm not a programmer, but there is a millions of those URL encoding-decoding wepages, so I think, that it is not that hard to put something like that, in the code, and voila! Bug fixed.

Lot of people already reported this same bug, so hopefully Vazkii give us an update with the fix soon.

**Comment by UntouchedWagons** [comment]
Yeah it's a major pain when downloading a mod pack with 240 mods in it. An alternate workaround is to download the server zip and use the mod files in that.
